lib/Controller/EndpointController.php

```diff
@@ -96,35 +97,29 @@ public function listNotifications() {
 			$data[] = $this->notificationToArray($notificationId, $notification);
 		}
 
-		return new \OC_OCS_Result(
-			$data,
-			100, // HTTP::STATUS_OK, TODO: <status>failure</status><statuscode>200</statuscode>
-			null,
-			['ETag' => $this->generateEtag($notificationIds)]
-		);
+		return new DataResponse($data, Http::STATUS_OK, ['ETag' => $this->generateEtag($notificationIds)]);
```

Comment:
use $respone->setETag
since that is used by the appframework to determine if the etag is the same.